IdentityCore/src/webview/operations/MSIDWebResponseBrokerInstallOperation.m

```diff
@@ -0,0 +1,104 @@
+
+- (nullable instancetype)initWithResponse:(nonnull MSIDWebviewResponse *)response
+                                    error:(NSError * _Nullable *)error
+{
+    #if TARGET_OS_IPHONE
+        self = [super initWithResponse:response
+                                 error:error];
+        if (self)
+        {
+            if (![response isKindOfClass:MSIDWebWPJResponse.class] || [NSString msidIsStringNilOrBlank:[(MSIDWebWPJResponse *)response appInstallLink]])
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"response is not valid");
+                *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Wrong type of response or response does not contain a valid app install link ", nil, nil, nil, nil, nil, YES);
+                return nil;
+            }
+            
+            MSIDWebWPJResponse *wpjResponse = (MSIDWebWPJResponse *)response;
+            _appInstallLink = [NSURL URLWithString:wpjResponse.appInstallLink];
+        }
+        
+        return self;
+    #else
+        return nil;
+    #endif
+}
+
+- (void)invokeWithInteractiveTokenRequestParameters:(nonnull MSIDInteractiveRequestParameters *)interactiveTokenRequestParameters
+                               tokenRequestProvider:(nonnull id<MSIDTokenRequestProviding>)tokenRequestProvider
+                                         completion:(nonnull MSIDRequestCompletionBlock)completion
+{
+    #if TARGET_OS_IPHONE
+        if ([interactiveTokenRequestParameters isKindOfClass:MSIDInteractiveTokenRequestParameters.class])
+        {
+            NSError *brokerError;
+            MSIDBrokerInteractiveController *brokerController = [[MSIDBrokerInteractiveController alloc] initWithInteractiveRequestParameters:(MSIDInteractiveTokenRequestParameters *)interactiveTokenRequestParameters
+                                                                                                                         tokenRequestProvider:tokenRequestProvider
+                                                                                                                            brokerInstallLink:self.appInstallLink
+                                                                                                                                        error:&brokerError];
+            if (brokerError)
+            {
+                MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"%@", [@"Not able to create brokerController" stringByAppendingString:[NSString stringWithFormat:@"%ld", (long)brokerError.code]]);
+                completion(nil, brokerError);
+                return;
+            }
+            
+            [brokerController acquireToken:completion];
+            return;
+        }
+    
+        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"interactiveTokenRequestParameters is in wrong type");
+        NSError *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Wrong type of interactive request parameter", nil, nil, nil, nil, nil, YES);
```

Comment:
nit: if you're calling MSID_LOG explicitly, no need to duplicate logging on error creation

Reply:
Removed places where both exists